### PR TITLE
MySql Provider

### PR DESCRIPTION
**Pull request by thefringeninja**
implements #29

### CODE REVIEW

File: tests/SqlStreamStore.TestUtils/MySql/MySqlDatabaseManager.cs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+                using(var connection = new MySqlConnection(DefaultConnectionString))
+                {
+                    connection.Open();
+
+                    using(var command = new MySqlCommand($"DROP DATABASE {DatabaseName}", connection))
+                    {
+                        command.ExecuteNonQuery();
+                    }
+
+                    using(var command = new MySqlCommand(
```
Comment (thefringeninja) [inline tests/SqlStreamStore.TestUtils/MySql/MySqlDatabaseManager.cs]: This is a huge WTF right here. After enough tests are run, tests begin to fail with:
```
MySql.Data.MySqlClient.MySqlException : Too many connections
Stack Trace:
   at MySqlConnector.Core.ServerSession.ConnectAsync(ConnectionSettings cs, ILoadBalancer loadBalancer, IOBehavior ioBehavior, CancellationToken cancellationToken) in C:\projects\mysqlconnector\src\MySqlConnector\Core\ServerSession.cs:line 290
   at MySqlConnector.Core.ConnectionPool.GetSessionAsync(MySqlConnection connection, IOBehavior ioBehavior, CancellationToken cancellationToken) in C:\projects\mysqlconnector\src\MySqlConnector\Core\ConnectionPool.cs:line 112
   at MySqlConnector.Core.ConnectionPool.GetSessionAsync(MySqlConnection connection, IOBehavior ioBehavior, CancellationToken cancellationToken) in C:\projects\mysqlconnector\src\MySqlConnector\Core\ConnectionPool.cs:line 141
   at MySql.Data.MySqlClient.MySqlConnection.CreateSessionAsync(Nullable`1 ioBehavior, CancellationToken cancellationToken) in C:\projects\mysqlconnector\src\MySqlConnector\MySql.Data.MySqlClient\MySqlConnection.cs:line 515
   at MySql.Data.MySqlClient.MySqlConnection.OpenAsync(Nullable`1 ioBehavior, CancellationToken cancellationToken) in C:\projects\mysqlconnector\src\MySqlConnector\MySql.Data.MySqlClient\MySqlConnection.cs:line 236
   at SqlStreamStore.MySqlStreamStore.OpenConnection(CancellationToken cancellationToken) in /repo/src/SqlStreamStore.MySql/MySqlStreamStore.cs:line 246
   at SqlStreamStore.MySqlStreamStore.CreateSchemaIfNotExists(CancellationToken cancellationToken) in /repo/src/SqlStreamStore.MySql/MySqlStreamStore.cs:line 74
   at SqlStreamStore.MySqlStreamStoreFixture.Init() in /repo/tests/SqlStreamStore.MySql.Tests/MySqlStreamStoreFixture.cs:line 53
   at SqlStreamStore.MySqlStreamStoreFixture.Create(ITestOutputHelper testOutputHelper, Boolean createSchema) in /repo/tests/SqlStreamStore.MySql.Tests/MySqlStreamStoreFixture.cs:line 63
   at SqlStreamStore.MySqlStreamStoreAcceptanceTests.CreateFixture() in /repo/tests/SqlStreamStore.MySql.Tests/MySqlStreamStoreAcceptanceTests.cs:line 23
   at SqlStreamStore.AcceptanceTests.InitializeAsync() in /repo/tests/SqlStreamStore.AcceptanceTests/AcceptanceTests.cs:line 30
```
I believe this is related to mysql-net/MySqlConnector#495 and mysql-net/MySqlConnector#469.

Eventually the server closes the connections so it isn't _completely_ broken. However this should be resolved before anyone attempts to use it in production.

**Comment by damianh** [comment]
Whoa!

**Comment by damianh** [comment]
The tests are passing ✨ Going to run it several times to check the stability. If good, gonna merge and code-review on master.

Nice PR lad.

**Comment by damianh** [comment]
https://travis-ci.org/SQLStreamStore/SQLStreamStore/builds/516930440#L1957

This seems to happen sometimes. It's probably a race, will need to look into it.

**Comment by damianh** [comment]
Gonna review this on master.

**Pull request merged**